### DISCUSSION TITLE
unhelpful `You have an error in your yaml syntax on line`

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#18629] [Actions]
## Unhelpful

So, here's a run https://github.com/electron/electron/actions/runs/2498900799

> Invalid workflow file: .github/workflows/release_dependency_versions.yml#L11
You have an error in your yaml syntax on line 11
>
> [Invalid workflow file: .github/workflows/release_dependency_versions.yml#L11](https://github.com/electron/electron/actions/runs/2498900799/workflow)
You have an error in your yaml syntax on line 11

This is not a helpful error message.

## GH Workflow Editor

Here's what the file looks like in GitHub's editor:
![image](https://user-images.githubusercontent.com/2119212/173709554-9770b8ac-05d3-4ad4-82e3-118a5264afa6.png)

It blames _line 14_ and says `bad indentation of a mapping entry`

![image](https://user-images.githubusercontent.com/2119212/173709623-04db68eb-0de0-47ac-b6b3-68bf73d40c26.png)

### Iterating

Adding a missing `steps:` item yields:
![image](https://user-images.githubusercontent.com/2119212/173709696-eba9f717-f902-4e22-a7ad-bd773c7930b4.png)

And removing the `id:` item entirely seems to make the workflow editor happy.

## Expected results

It'd be great if the error returned at the top matched the error returned by the workflow editor:

> Invalid workflow file: .github/workflows/release_dependency_versions.yml#L14
Bad indentation in your yaml file on line 14

**Comment by MathiasReker** [comment]
Uses must be after steps. Check this example: https://github.com/MathiasReker/CI-CD/blob/develop/.github/workflows/ci-cd.yml

**Reply by jsoref** [reply]
I know. That isn't the point. The point is that the error message is unhelpful.
